d2l-sort-by-option.html

```diff
@@ -0,0 +1,46 @@
+<link rel="import" href="../polymer/polymer-element.html">
+<link rel="import" href="../d2l-dropdown/d2l-dropdown-menu.html">
+<link rel="import" href="../d2l-dropdown/d2l-dropdown-button-subtle.html">
+<link rel="import" href="../d2l-menu/d2l-menu-item.html">
+<link rel="import" href="../d2l-menu/d2l-menu-item-behavior.html">
+
+<dom-module id="d2l-sort-by-option">
+	<template strip-whitespace>
+		<style>
+			:host {
+				display: flex;
+				flex-direction: column;
+			}
+			d2l-menu-item {
+				/**
+				 * Needed to show the background colour of d2l-menu-item when hovering;
+				 * d2l-menu-item is overriding the :hover style of d2l-menu
+				 */
+				background-color: transparent;
+			}
+		</style>
+		<d2l-menu-item text="[[text]]"></d2l-menu-item>
+	</template>
+	<script>
+		/**
+		 * `<d2l-sort-by-option>`
+		 * Option for sort by dropdown
+		 */
+		class SortByOption extends Polymer.mixinBehaviors(
```

Comment:
Similar to my other comment in the "close" thread, it is very strange that this `SortByOption` component be a menu-item (i.e. by including MenuItemBehavior), which itself has a `d2l-menu-item` as a direct child.  `d2l-menu` should be the only direct parent of `role="menuitem"`.  This may break a11y and other menu behaviors.  The only time we should have menu items inside of menu items is for nested menus, where there is `d2l-menu` > `role="menuitem"` > `d2l-menu` > `role="menuitem"` ... etc.

Reply:
Yeah my attempted fix resulted in this basically becoming a custom menu item

The intent was to keep the elements named as they are, but have the same behaviour as the `d2l dropdown-menu`. It seems like it might cause more issues than it solves

Given this, would you/@dlockhart suggest changing the structure of this component to require a `d2l-dropdown` to be used?

Reply:
Yep - that's certainly doable.  The majority of the logic is for menu items and selectable menu items is already in a behavior, so it would just be a little bit of radio-specific logic that gets moved.  So there would be a new behavior along these lines:
```
D2L.PolymerBehaviors.MenuItemRadioBehaviorImpl = {
	...   
};

D2L.PolymerBehaviors.MenuItemRadioBehavior = [
	D2L.PolymerBehaviors.MenuItemRadioBehaviorImpl,
	D2L.PolymerBehaviors.MenuItemSelectableBehavior
];
```
Then, in a similar way as this [custom menu-item demo](https://github.com/BrightspaceUI/menu/blob/master/demo/demo-components.html#L8), a sort-specific selectable menu item could be defined (using the new radio behavior).

Reply:
Sounds good, I'll work on creating the `MenuItemRadioBehavior`